src/core/Akka.Tests/Actor/ActorLookupSpec.cs

```diff
@@ -0,0 +1,179 @@
+
+            public IEnumerable<string> Path { get; private set; }
+        }
+
+        public sealed class LookupString : IQuery
+        {
+            public LookupString(string path)
+            {
+                Path = path;
+            }
+
+            public string Path { get; private set; }
+        }
+
+        public sealed class LookupPath : IQuery
+        {
+            public LookupPath(ActorPath path)
+            {
+                Path = path;
+            }
+
+            public ActorPath Path { get; private set; }
+        }
+
+        public sealed class GetSender : IQuery
+        {
+            public GetSender(IActorRef to)
+            {
+                To = to;
+            }
+
+            public IActorRef To { get; private set; }
+        }
+
+        public static readonly Props P = Props.Create<Node>();
+
+        public class Node : ReceiveActor
+        {
+            private IActorRefProvider Provider { get { return Context.System.AsInstanceOf<ActorSystemImpl>().Provider; } }
+
+            public Node()
+            {
+                Receive<Create>(create => Sender.Tell(Context.ActorOf(P, create.Child)));
+                Receive<LookupElems>(elems => Sender.Tell(Provider.ResolveActorRef(new RootActorPath(Provider.DefaultAddress) / elems.Path)));
+                Receive<LookupPath>(path => Sender.Tell(Provider.ResolveActorRef(path.Path)));
+                Receive<LookupString>(str => Sender.Tell(Provider.ResolveActorRef(str.Path)));
+                Receive<GetSender>(s => s.To.Tell(Sender));
+            }
+        }
+
+        private IActorRef c1;
+        private IActorRef c2;
+        private IActorRef c21;
+
+        protected readonly ActorSystemImpl SysImpl;
+        protected IActorRefProvider Provider => SysImpl.Provider;
+
+        private IActorRef user;
+        private IActorRef syst;
+        private IActorRef root;
+
+        public ActorLookupSpec()
+        {
+            c1 = Sys.ActorOf(P, "c1");
+            c2 = Sys.ActorOf(P, "c2");
+            c21 = c2.Ask<IActorRef>(new Create("c21"), RemainingOrDefault).Result;
+            SysImpl = (ActorSystemImpl) Sys;
+            user = SysImpl.Guardian;
+            syst = SysImpl.SystemGuardian;
+            root = SysImpl.LookupRoot;
+        }
+
+        private IActorRef Emtpy(string path)
+        {
+            return new EmptyLocalActorRef(SysImpl.Provider, root.Path / path, Sys.EventStream);
+        }
+
+        [Fact]
+        public void ActorSystem_must_find_actors_by_looking_up_their_path()
+        {
+            Provider.ResolveActorRef(c1.Path).Should().Be(c1);
+            Provider.ResolveActorRef(c2.Path).Should().Be(c2);
+            Provider.ResolveActorRef(c21.Path).Should().Be(c21);
+            Provider.ResolveActorRef(user.Path / "c1").Should().Be(c1);
+            Provider.ResolveActorRef(user.Path / "c2").Should().Be(c2);
+            Provider.ResolveActorRef(user.Path / "c2" / "c21").Should().Be(c21);
```

Comment:
I skipped those because `ResolveActorRef` has different semantics than `ActorFor` - `ActorSelection` handles this type of functionality now. `ResolveActorRef` only works for absolute lookups.